libs/filamat/README.md

```diff
@@ -170,3 +173,48 @@ material with Filament, pass the material package's data into a Filament Materia
 
 Note that this will require [linking against Filament's libraries](../../filament/README.md) in
 addition to Filamat's.
+
+## Filamat Lite
+
+The `filamat_lite` library is interchangeable with `filamat`, with a few caveats:
+
+1. Material compilation is only supported for the OpenGL backend.
+1. No shader-level optimization is performed.
+1. GLSL correctness is not checked.
+
+In addition, `filamat_lite` only performs a simple text match to determine which properties on the
+`MaterialInputs` structure are set. It does this by looking for instances of `material.`. Because of
+this, shaders must be careful with whitespace and comments. The `material` input variable must also
+always be refered to by the name `material`.
+
+```
+void anotherFunction(inout MaterialInputs m) {
+    // Incorrect! The MaterialInputs is being referred to by the name "m".
+    m.metallic = 0.0;
+}
+
+void aFunction(inout MaterialInputs material) {
+    // Works, but only because the variable name "material" is used.
+    material.reflectance = 0.5;
+}
+
+// The MaterialInputs variable must be named material.
+void material(inout MaterialInputs material) {
+    prepareMaterial(material);
+
+    // Good.
+    material.roughness = materialParams.roughness;
+    material.baseColor.rgb = vec3(1.0, 0.0, 1.0);
+
+    // Incorrect! While this line is technically correct GLSL code, the simple parser inside
+    // filamat_lite will not pick up the set to clearCoat due to the additional whitespace.
```

Comment:
Sounds like something we could easily support, although probably not worth it

Reply:
Yeah, I originally had a `regex` which would handle this case, but `std::regex` pulls in ~40KiB or so.. I decided it wasn't worth it as it's probably an edge case